### PR TITLE
Add perturbation features in Fortran/pp_ser

### PR DESCRIPTION
**Pull request by clementval**
Implementation of the perturbation feature for #48. 

The perturbation module `m_pertub` is an extension from the module found in `cosmo-pompa` with single/double precision as well as missing dimensions. 

In addition, I add a small example to use Serialbox from fortran with three mode: read, read-perturb and write. So it kind of solve #15 
The example is located in `example/perturbation` and there is a full `cmake` file to build it as a standalone app as well as a README to explain it.

### CODE REVIEW

**Comment by cosunae** [comment]
since the main core of the serialbox is C++ with fortran + python wrapper, and perturbation directly touches storages, I would be in favor of implementing this functionality in C++, and just call it from fortran.
Also since the SerialBox is supposed to be more general than cosmo, I would allow to pass a user functor so that the user can customize how to perturb the data. Otherwise a hardcoded fomula for the perturbation does not look general enough for serialbox.

**Comment by pspoerri** [comment]
screen shot 2016-07-15 at 14 50 34

I agree, but we also have a limited time available for Serialbox stuff. Additionally the problem is that in C++ we work with the raw data. 
I'm currently in favor of integrating it in Fortran for the time being.

To allow for later upgrades we could do the naming in such a way that we can simply exchange the interface some time in the future.

File: examples/perturbation/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ADD_CUSTOM_COMMAND(
+  OUTPUT pp/main_consumer_perturb.f90
+  COMMAND ${CMAKE_SOURCE_DIR}/../../python/pp_ser.py -d pp ${CMAKE_SOURCE_DIR}/main_consumer_perturb.f90
+  DEPENDS main_consumer_perturb.f90
+)
+ADD_CUSTOM_COMMAND(
+  OUTPUT pp/main_producer.f90
+  COMMAND ${CMAKE_SOURCE_DIR}/../../python/pp_ser.py -d pp ${CMAKE_SOURCE_DIR}/main_producer.f90
+  DEPENDS main_producer.f90
+)
```
Comment (pspoerri) [inline examples/perturbation/CMakeLists.txt:33]: Can't we use one custom command? https://cmake.org/cmake/help/v3.0/command/add_custom_command.html

**Comment by pspoerri** [comment]
looks good to me.

File: examples/perturbation/CMakeLists.txt (hunk 1 above)
Comment (clementval) [inline examples/perturbation/CMakeLists.txt:33]: Yeah you are right ... I'll give it a shot

**Comment by clementval** [comment]
@cosunae Well as @pspoerri mentioned, in C++ we have raw data so it's harder to applied "simply" a perturbation. The goal here was to have the feature with minimum effort as it was requested for quite long time by @lxavier. So in my opinion, we can discuss to change the feature when we really separate serialbox/cosmo features and when we have more time or someone to work on it.

**Comment by clementval** [comment]
So if no other feedback, I'll merge this today.

**Comment by lxavier** [comment]
great ! thanks for the feature !

**Comment by clementval** [comment]
launch jenkins

All passed besides Greina ...

**Pull request merged**